src/rgw/rgw_sal.cc

```diff
@@ -117,7 +117,7 @@ rgw::sal::Store* StoreManager::init_storage_provider(const DoutPrefixProvider* d
       ldpp_dout(dpp, 0) << "newMotrStore() failed!" << dendl;
       return store;
     }
-    ((rgw::sal::MotrStore *)store)->init_metadata_cache(dpp, cct);
+    ((rgw::sal::MotrStore *)store)->init_metadata_cache(dpp, cct, use_cache);
```

Comment:
From where exactly we are setting up value of `use_cache`?

Reply:
in ceph.conf, we can configure it.
the default value is true.
```
[client]
   .
   rgw_cache_enabled = false
   .
   .
```

Reply:
I meant where exactly `use_cache` is getting initialized with value from `rgw_cache_enabled` in case of store being motr.

Reply:
In rgw_main.cc, while initializing the Store, we fetch it from the config and pass it as an argument
```c++
  rgw::sal::Store* store =
    StoreManager::get_storage(&dp, g_ceph_context,
		 rgw_store,
		 g_conf()->rgw_enable_gc_threads,
		 g_conf()->rgw_enable_lc_threads,
		 g_conf()->rgw_enable_quota_threads,
		 g_conf()->rgw_run_sync_thread,
		 g_conf().get_val<bool>("rgw_dynamic_resharding"),
		 g_conf()->rgw_cache_enabled);
```